### PR TITLE
Refactored Equals And HashCode for entities.

### PR DESCRIPTION
**Pull request by ivanCosic**
Fixed a pending invitation bug found while testing hotfix.

### CODE REVIEW

File: repository/src/main/java/eu/execom/labs/pomodorospringreact/repository/PendingInvitationDao.java [hunk 1]
```diff
@@ -49,6 +51,10 @@ public PendingInvitation getPendingInvitation(long userId, long teamId) {
                 .setParameter(1, userId)
                 .setParameter(2, teamId);
 
-        return (PendingInvitation) query.getSingleResult();
+        try {
```
Comment (idostanic) [inline repository/src/main/java/eu/execom/labs/pomodorospringreact/repository/PendingInvitationDao.java]: this does not have to do anything with hashcode and equals...

File: repository/src/main/java/eu/execom/labs/pomodorospringreact/entity/User.java [hunk 2]
```diff
@@ -21,7 +21,7 @@
 @NoArgsConstructor
 @Getter
 @Setter
-@EqualsAndHashCode(of = {"email"})
+@EqualsAndHashCode(exclude = {"id", "pomodoros", "userTeams", "pendingInvitations"})
```
Comment (idostanic) [inline repository/src/main/java/eu/execom/labs/pomodorospringreact/entity/User.java]: why are the other fields excluded?

File: repository/src/main/java/eu/execom/labs/pomodorospringreact/repository/PendingInvitationDao.java (hunk 1 above)
Comment (ivanCosic) [reply repository/src/main/java/eu/execom/labs/pomodorospringreact/repository/PendingInvitationDao.java]: When i tried to test equals&hashcode refactor, I run into exception caused by this, and I've decided to fix it in order to be able to test equals&hashcode.

File: repository/src/main/java/eu/execom/labs/pomodorospringreact/entity/User.java (hunk 2 above)
Comment (ivanCosic) [reply repository/src/main/java/eu/execom/labs/pomodorospringreact/entity/User.java]: If I include collections, it tries to calculate hascode for each entity, which in return tries to callculate hashcode for it's connected entities, and many recursive calls happen until stack overflow happens. I've excluded all collections from calculating of equals&hashcode.

**Pull request merged**